### PR TITLE
report an event to pod if kubelet does attach operation failed

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

kubelet is running with --enable-controller-attach-detach=false. 

kubelet log:

```
May 15 09:55:17 demo-control-plane kubelet[691]: E0515 09:55:17.993787     691 nestedpendingoperations.go:348] Operation for "{volumeName:kubernetes.io/csi/hostpath.csi.k8s.io^fa71a846-129d-11ef-a424-4a1499c02f41 podName: nodeName:}" failed. No retries permitted until 2024-05-15 09:57:19.993627526 +0000 UTC m=+3002.459216475 (durationBeforeRetry 2m2s). Error: AttachVolume.Attach failed for volume "pvc-d8752815-33f2-4609-bf8f-9ca004058dd6" (UniqueName: "kubernetes.io/csi/hostpath.csi.k8s.io^fa71a846-129d-11ef-a424-4a1499c02f41") from node "demo-control-plane" : attaching volumes from the kubelet is not supported
```

no event:

```
(base) (⎈|kind-demo:N/A)➜  ~ kubectl get po
NAME                   READY   STATUS              RESTARTS   AGE
csi-hostpath-socat-0   1/1     Running             0          85m
csi-hostpathplugin-0   8/8     Running             0          72m
my-csi-app             0/1     ContainerCreating   0          47m
(base) (⎈|kind-demo:N/A)➜  ~ kubectl describe po my-csi-app
...
Events:
  Type    Reason     Age   From               Message
  ----    ------     ----  ----               -------
  Normal  Scheduled  47m   default-scheduler  Successfully assigned default/my-csi-app to demo-control-plane
```

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
report an event to pod if kubelet does attach operation failed when kubelet is running with `--enable-controller-attach-detach=false`
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/volumemanager/reconciler/reconciler_common.go [hunk 1]
```diff
@@ -250,9 +250,10 @@ func (rc *reconciler) waitForVolumeAttach(volumeToMount cache.VolumeToMount) {
 		// Volume is not attached to node, kubelet attach is enabled, volume implements an attacher,
```
Comment (carlory) [inline pkg/kubelet/volumemanager/reconciler/reconciler_common.go:250]: Is it legacy behavior? Should we deprecate and remove it in a future release?

Comment (jsafrane) [reply pkg/kubelet/volumemanager/reconciler/reconciler_common.go:250]: Yes, it's very legacy. I doubt anyone uses this code.

**Comment by jsafrane** [comment]
I think nobody uses this code path, still, reporting an event won't harm
/lgtm
/approve

File: pkg/kubelet/volumemanager/reconciler/reconciler_common.go (hunk 1 above)
Comment (carlory) [reply pkg/kubelet/volumemanager/reconciler/reconciler_common.go:250]: According to https://github.com/kubernetes/kubernetes/issues/20262#issue-129377066

Kubelet volume attach/detach enabled, this mode is primarily provided to support two scenarios:

- Support rolling upgrades of Kubernetes 
- To support volumes defined in static pods defined on standalone (no master) kubelets.

The following plugins implement the attach/detach interface:

- csi (don't support kubelet volume attach/detach)
- fc (attach is noop, attach logic is in WaitForAttach)
- flexvolume (deprecated but still available, flex may support both attach/detach in kubelet or controller)
- iscsi (attach is noop, attach logic is in WaitForAttach)

So we only take care of the flexvolume plugin. Once the flexvolume plugin is removed, the kubelet attach/detach feature can be removed as well.

The flag `--enable-controller-attach-detach` is not marked as deprecated. Can we mark it as deprecated in the next release? @jsafrane

**Pull request merged**